service_catalog/mail_utils.py

```diff
@@ -37,36 +25,61 @@ def _get_headers(subject):
     return headers
 
 
-def _get_admin_emails(object_to_filter):
-    """
-    Return a list of admin (is_staff) email if notification is enabled and target service subscribed
-    :return:
-    """
-    admins = User.objects.filter(is_staff=True)
-    # create a list of email
-    email_admins = list()
-    for admin in admins:
-        if admin.email:
-            if isinstance(object_to_filter, Instance) and admin.profile.instance_notification_enabled:
-                if admin.profile.is_notification_authorized_for_instance(object_to_filter):
-                    email_admins.append(admin.email)
-            elif isinstance(object_to_filter, Request) and admin.profile.request_notification_enabled:
-                if admin.profile.is_notification_authorized_for_request(object_to_filter):
-                    email_admins.append(admin.email)
-    return email_admins
+def _apply_when_filter_instance(user_list, squest_object):
+    users = list()
+    for user in user_list:
+        if user.email:
+            if user.profile.instance_notification_enabled:
```

Comment:
So, an User queryset can be filtered easly

Reply:
You're right, something like
```python
return qs.filter(email__isnull=False,profile__notification_filter_enabled=True) 
```